### PR TITLE
Years Select

### LINKED ISSUE #56: Select for Years
- Right now it is hardcoded to 1 - 5

Possible Options
- change options available based on onchange of program
- populate with years available from any course
- remain hardcoded but done with enums

### PR DESCRIPTION
**Pull request by K-shamG**
Changed year select to update when program is selected (some classes don't apply to all years)

Closes #56

### CODE REVIEW

**Review by andrewdlqnguyen — APPROVED** [review]
Looks good

File: src/main/java/SYSC4806/Controller/HomeController.java [hunk 1]
```diff
@@ -101,6 +104,17 @@ public String showResults(Model model, @PathVariable("program") String p, @PathV
         return "fragments/results :: resultsTable";
     }
 
+    @GetMapping("/results/{program}")
+    public ResponseEntity<Object> getYears(@PathVariable("program") String p) {
```
Comment (RyanWork) [inline src/main/java/SYSC4806/Controller/HomeController.java:112]: Looks okay, can you just add some documentation on this method?

File: src/main/resources/static/js/index.js [hunk 2]
```diff
@@ -1,10 +1,34 @@
...
     $('#results-table').DataTable();
+
+    // Starting jQuery library select2 for searchable select boxes https://select2.org/
+    $('.select2').select2({
+        placeholder: 'Choose...',
+        allowClear: true
+    });
 });
 
+$('#programSelect').change( function() {
```
Comment (RyanWork) [inline src/main/resources/static/js/index.js:15]: Can you also add some documentation just saying what you're binding this event-handler to. I don't think it's entirely intuitive just from looking at what's going on in here.

**Review by RyanWork — CHANGES_REQUESTED** [review]
Functions well :) Just some dev comments please!

**Pull request merged**